File: Processus/exercice2.3.py
import os
import sys

def main():
    for i in range(5):
        try:
            pid = os.fork()
        except OSError:
            print("Erreur lors de la création du processus")
            sys.exit(1)

        if pid == 0:
            print(f"Je suis le fils {i + 1}")
            break

if __name__ == "__main__":
    main()

File: Processus/exercice2.3.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>

int main()
{
    int i ;
    pid_t pid ;
    for(i = 0 ; i < 5; i++)
    {
        pid = fork() ;
        if(pid == -1)
        {
            printf("Erreur lors de la création du processus ") ;
            exit(1) ;
        }
        else if(pid == 0)
        {
            printf("Je suis le fils %d \n", i+1) ;
            break ;
        }
    }
    return 0;

        
}